src/org/labkey/targetedms/parser/skyaudit/AuditLogEntry.java

```diff
@@ -246,6 +251,36 @@ public AuditLogEntry persist(){
         return this;
     }
 
+    /**
+     * Updates entry's versionId field in the database. It is the only field that can change,
+     * the rest of the entry is immutable.
+     * @param pVersionId new versionId value.
+     * @return this instance if the update is successful.
+     */
+    public AuditLogEntry updateVersionId(Integer pVersionId){
+        this.setVersionId(pVersionId);
+        SQLFragment sqlUpdate = new SQLFragment("");
+
+        if(this._entryId == null){  //entryId can be null if this is an existing entry read from a file and not persisted into the database
+            //attempting to retrieve entryId using the alternative key of document GUID and entry hash.
+            sqlUpdate.append("UPDATE targetedms.AuditLogEntry SET versionId = ? WHERE documentGUID = ? AND entryHash = ?");
+            sqlUpdate.addAll(List.of(pVersionId, this._documentGUID.toString(), this._entryHash));
```

Comment:
Don't think the List.of() should be required - can just call the vararg variant of addAll() directly